openedx/core/djangoapps/notifications/utils.py

```diff
@@ -121,19 +121,19 @@ def filter_out_visible_notifications(
     :param user_forum_roles: List of forum roles for the user
     :return: Updated user preferences dictionary
     """
-    for key in user_preferences:
-        if 'notification_types' in user_preferences[key]:
-            # Iterate over the types to remove and pop them from the dictionary
-            for notification_type, is_visible_to in notifications_with_visibility.items():
-                is_visible = False
-                for role in is_visible_to:
-                    if role in user_forum_roles:
-                        is_visible = True
-                        break
-                if is_visible:
-                    continue
-
-                user_preferences[key]['notification_types'].pop(notification_type)
+    discussion_user_preferences = user_preferences.get('discussion', {})
+    if 'notification_types' in discussion_user_preferences:
```

Comment:
Why are we only filtering `discussion` app notifications? I think filters should be applied to all notification types.

Reply:
the only visibility filter we currently have are on forum roles, and restricting them to discussions app makes sense as no other app will restrict based on forum roles!

Reply:
We can revisit this when more visibility filters are need, as we would need a more generic function for that. For now it makes sense to restrict this to discussion notifications. 
Let me know your thoughts?